sdk/data_tables/src/operations/get_entity.rs

```diff
@@ -0,0 +1,105 @@
+use crate::prelude::*;
+use azure_core::{
+    collect_pinned_stream, headers::*, AppendToUrlQuery, Context, Etag, Method, Response,
+};
+use azure_storage::core::headers::CommonStorageResponseHeaders;
+use serde::de::DeserializeOwned;
+use std::convert::TryInto;
+
+#[derive(Debug, Clone)]
+pub struct GetEntityBuilder {
+    entity_client: EntityClient,
+    select: Option<Select>,
+    context: Context,
+}
+
+impl GetEntityBuilder {
+    pub(crate) fn new(entity_client: EntityClient) -> Self {
+        Self {
+            entity_client,
+            select: None,
+            context: Context::new(),
+        }
+    }
+
+    setters! {
+        select: Select => Some(select),
+        context: Context => context,
+    }
+
+    pub fn into_future<E>(mut self) -> FutureResponse<E>
+    where
+        E: DeserializeOwned,
+    {
+        Box::pin(async move {
+            let mut url = self.entity_client.url().to_owned();
+
+            self.select.append_to_url_query(&mut url);
+
+            let mut headers = Headers::new();
+            headers.insert(ACCEPT, "application/json;odata=fullmetadata");
+
+            let mut request =
+                self.entity_client
+                    .finalize_request(url, Method::Get, headers, None)?;
+
+            let response = self
+                .entity_client
+                .send(&mut self.context, &mut request)
+                .await?;
+
+            GetEntityResponse::try_from(response).await
+        })
+    }
+}
+
+pub type FutureResponse<E> =
+    futures::future::BoxFuture<'static, azure_core::Result<GetEntityResponse<E>>>;
+
+#[cfg(feature = "into_future")]
+impl std::future::IntoFuture for GetEntityBuilder {
+    type IntoFuture = FutureResponse<E>;
```

Comment:
This doesn't compile. The Builder should be made generic. See `CreateDocument` in cosmos db.